### ISSUE TITLE
Refactor model test

### ISSUE THREAD

**Issue by swimdi** [tenstorrent/pytorch2.0_ttnn#269]
Current model test have a torch_ttnn model run if set torch_ttnn property, but as the training mode join, this run_model function become insufficient
```
# in training mode it should need to run backward to collect the backward ops
def run_model(model, inputs):
    if isinstance(inputs, collections.Mapping):
        return model(**inputs)
    elif isinstance(inputs, collections.Sequence):
        return model(*inputs)
    else:
        return model(inputs)
```

So there propose to refactor the model test, let it have a `ModelTester` class and handle the run model flow
